ports/stm32/Makefile

```diff
@@ -417,6 +417,13 @@ endif
 
 endif # MICROPY_PY_BLUETOOTH
 
+ifeq ($(MICROPY_PY_OPENAMP),1)
+SRC_C += metal/metal_port.c
```

Comment:
Suggest this directory is renamed `openamp` (or `openamp/metal`), even though it currently only contains a metal port.

The reason is to avoid developers needing to be familiar with two openamp-associated names at the port level. From our perspective `metal` is part of the implementation of `openamp`, so it's easier to understand it if it's scoped there first. (Currently most of the "high level" names are `openamp` already, I think it's just this metal/ subdirectory.)

This may seem like a nitpick, but Micropython is a big project so it's useful to keep the naming clean whenever possible.

Reply:
No it's good suggestion, if you want to override something else or maybe provide a new image store, this will be the right place to add it.

Reply:
To add some more context on this... my understanding was that we only created subdirectories in ports when need to match an existing project's include path. i.e. if the third-party project does `#include "foo/config.h"` then we need do be able to pass something to `-I` that contains `foo`. (e.g. lwip, which expects to be able to do `#include "arch/cc.h", which is why we "hide" it inside the "lwip_inc" dir).

In this case, we actually control the include, so it can just be "mpmetalport.h" as in the commit below (or "mpopenampport_libmetal.h" might be better).

Reply:
This is why I originally added it to `metal/metal_config.h` because port dir is included in the path by default, and the library files included `metal/x.h` so I wanted to match the include paths in the library. I prefer `ports/<port>/mpmetalport.h/c` just to avoid making the file name any longer than necessary.

Reply:
Fair enough, thanks @jimmo for the extra explanation.